stdlib/solana.c

```diff
@@ -82,19 +82,28 @@ uint64_t external_call(uint8_t *input, uint32_t input_len, SolParameters *params
         .data_len = input_len,
     };
 
+    int meta_no = 1;
+
     for (int account_no = 0; account_no < params->ka_num; account_no++)
     {
         const SolAccountInfo *acc = &params->ka[account_no];
 
         if (SolPubkey_same(dest, acc->key))
         {
+            metas[0].pubkey = acc->key;
+            metas[0].is_writable = acc->is_writable;
+            metas[0].is_signer = acc->is_signer;
```

Comment:
Don't we need tests for these modifications?

Reply:
So the input for the solana program is now:
 - data account for contract
 - data account for sender (=broken)
 - value sent (no longer used)
 - seeds (soon to be removed)
 - hash of contract name to be called (soon to be removed)
 - abi encoded stuff

This PR now also passes the data account via the accounts, and the account must the first. This is now enforced on this line:

https://github.com/hyperledger/solang/pull/1039/files#diff-3b88924fae67d4d137158f036752a5f2b28f3fe12d968fc17c49f0ff7311b110R54

This ensures that the data account passed via the accounts is the same as the first as the input. So, any test which runs on a VM tests this.

(Note that the aim is remove all this stuff from the input, and I think we are near to that).